src/metrics-gatherer.ts

```diff
@@ -112,6 +112,21 @@ export class MetricsGatherer {
 		}
 	}
 
+	// increment/decrement a gauge during the runtime of a function
+	public async run(name: string, func: () => any, labels: LabelSet = {}) {
+		this.inc(name, 1, labels);
+		await func();
+		this.inc(name, -1, labels);
+	}
+
+	// increment/decrement a gauge during the runtime of a promise (wait for it to resolve)
+	public resolve(name: string, promise: Promise<any>, labels: LabelSet = {}) {
+		this.inc(name, 1, labels);
+		promise.finally(() => {
+			this.inc(name, -1, labels);
+		});
```

Comment:
This is better as
```suggestion
		try {
			await promise
		} finally {
			this.inc(name, -1, labels);
		}
```
because it should also handle promises that don't support `.finally`